src/libopenrave/kinbodylink.cpp

```diff
@@ -337,21 +340,33 @@ void KinBody::LinkInfo::DeserializeJSON(const rapidjson::Value &value, dReal fUn
         }
     }
 
-    _mapExtraGeometries.clear();
-    // if (value.HasMember("extraGeometries")) {
-    //     for (rapidjson::Value::ConstMemberIterator it = value["extraGeometries"].MemberBegin(); it != value["extraGeometries"].MemberEnd(); ++it) {
-    //         if (_mapExtraGeometries.find(it->name.GetString()) == _mapExtraGeometries.end()) {
-    //             _mapExtraGeometries[it->name.GetString()] = std::vector<GeometryInfoPtr>();
-    //         }
-    //         std::vector<GeometryInfoPtr>& vgeometries = _mapExtraGeometries[it->name.GetString()];
-    //         vgeometries.reserve(it->value.Size() + vgeometries.size());
-    //         size_t iGeometry = 0;
-    //         for(rapidjson::Value::ConstValueIterator im = it->value.Begin(); im != it->value.End(); ++im, ++iGeometry) {
-    //             std::string id = orjson::GetStringJsonValueByKey(*im, "id");
-    //             UpdateOrCreateInfoWithNameCheck(*im, id, vgeometries, "name", fUnitScale);
-    //         }
-    //     }
-    // }
+    if (value.HasMember("extraGeometries") && value["extraGeometries"].IsArray()) {
+        for (rapidjson::Value::ConstValueIterator it_l = value["extraGeometries"].Begin(); it_l != value["extraGeometries"].End(); ++it_l) {
+            std::string name;
+            // has valid extra geometry
+            if (it_l->HasMember("name") && it_l->HasMember("id")) {
+                orjson::LoadJsonValueByKey(*it_l, "name", name);
+            }
+            else {
+                continue;
+            }
+
+            if (_mapExtraGeometries.find(name) == _mapExtraGeometries.end()) {
+                _mapExtraGeometries[name] = std::vector<GeometryInfoPtr>();
+            }
+
+            if (it_l->HasMember("geometries")) {
```

Comment:
Here we need to deal with merging incoming changes based on the `id` field instead of replacing the data directly.

Reply:
I modified the deserialization to merge incoming changes instead of replacing. Can you review the new commit?